metricflow/test/sql/optimizer/test_sub_query_reducer.py

```diff
@@ -266,3 +266,65 @@ def test_rewrite_order_by_with_a_join_in_parent(
         sql_plan_node=sub_query_reducer.optimize(rewrite_order_by_statement),
         plan_id="after_reducing",
     )
+
+
+def test_rewrite_distinct_select_node(
```

Comment:
Maybe `test_reduce_distinct_select_node`?

Also, I prefer the test name to show what should happen, like `test_distinct_select_node_is_not_removed` or something, but if that's not how the test files do their naming this is fine as-is.